### DISCUSSION TITLE
Install in qBittorrent?

### DISCUSSION THREAD

**Discussion by Zonimi** [jesec/flood#164] [Q&A]
Hi, I saw that you have a experimental version of qBittorrent, but I can't find anything about it to install it

**Comment by jesec** [comment]
Flood is a Web UI that supports qBittorrent. There is no "experimental" qBittorrent.

**Reply by Zonimi** [reply]
> Flood is a Web UI that supports qBittorrent. There is no "experimental" qBittorrent.

Yes, I mean the experimental version for qbit as it is on the page https://www.npmjs.com/package/flood, client> qbittorrent version> experimental.

And I'm having trouble starting the flood with qbittorrent, since I can't find any guides on it.

`/usr/local/lib/node_modules/flood/dist/index.js:2`
`module.exports=(()=>{var __webpack_modules__={3064:(e,t,i)=>{const{spawn:n}=i(63129);const r=i(76417);const a=i(35747);const o=i(12087);const s=i(85622);const{argv:c}=i(64139).env("FLOOD_OPTION_").option("baseuri",{default:"/",describe:"This URI will prefix all of Flood's HTTP requests",type:"string"}).option("rundir",{alias:"d",default:s.join(o.homedir(),".local/share/flood"),describe:"Where to store Flood's runtime files (eg. database)",type:"string"}).option("host",{alias:"h",default:"127.0.0.1",describe:"The host that Flood should listen for web connections on",type:"string"}).option("port",{alias:"p",default:3e3,describe:"The port that Flood should listen for web connections on",type:"number"}).option("secret",{alias:"s",hidden:true,describe:"A unique secret, a random one will be generated if not provided",type:"string"}).option("auth",{describe:"Access control and user management method",choices:["default","none"]}).option("noauth",{alias:"n",hidden:tr`

`Error: listen EADDRNOTAVAIL: address not available 34.105.100.23:8081`
    `at Server.setupListenHandle [as _listen2] (net.js:1263:19)`
    `at listenInCluster (net.js:1328:12)`
    `at doListen (net.js:1461:7)`
    `at process._tickCallback (internal/process/next_tick.js:63:19)`

**Reply by jesec** [reply]
Experimental means Flood's support of qBittorrent is experimental. There is no experimental build of qBittorrent itself.

**Reply by Zonimi** [reply]
> Experimental means Flood's support of qBittorrent is experimental. There is no experimental build of qBittorrent itself.

I know friend, version of the experimental Flood for qbittorrent.
And about my problem, do you know what it can be?

**Reply by jesec** [reply]
It appears that you are trying to make Flood listen on a host that it can't bind. Note that by default you configure torrent client (qBittorrent) connection settings in web interface. Read README carefully.

**Reply by Zonimi** [reply]
I followed the example in the readme.

`qbittorrent-nox --daemon --webui-port=8081 --profile=/root/testq`
`flood --rundir /root/testq --allowedpath /root/testq --auth none --qburl http://34.105.100.23:8081 --qbuser admin --qbpass adminadmin --assets false`

And I had this return:

```
Flood server starting on http://127.0.0.1:3000.

Starting without builtin authentication
```

But http://34.105.100.23:3000/ does not work.
(Yes, port is open)

**Reply by jesec** [reply]
By default, Flood listens on localhost only. 

You need to specify `—host 0.0.0.0` argument to listen on all interfaces. Additionally, don’t `—assets false` unless you configure other web servers to serve static assets. 

And, DO NOT blindly use parameters of tests. TESTS are not EXAMPLES. You can take inspiration from it, but those parameters are probably not good for production. For instance, you probably  don’t really want to disable authentication.

**Reply by Zonimi** [reply]
Oh yes, I was just trying to make it work before setting it up correctly, but now it's working, thanks.

**Reply by Zonimi** [reply]
> By default, Flood listens on localhost only.
> 
> You need to specify `—host 0.0.0.0` argument to listen on all interfaces. Additionally, don’t `—assets false` unless you configure other web servers to serve static assets.
> 
> And, DO NOT blindly use parameters of tests. TESTS are not EXAMPLES. You can take inspiration from it, but those parameters are probably not good for production. For instance, you probably don’t really want to disable authentication.

Hello again, I'm having another problem, after registering the account for Flood and updating the page I have a message that it was not possible to connect to the torrent client.

And when trying to establish the connection again, says that it was not possible

**Reply by jesec** [reply]
Well. Check the connection to your torrent client, then. Make sure that you got the configurations right.
